### PR TITLE
feature/5966-Dylan-FindLocationAnalytics

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Added analytics for when Find a VA location is clicked on home screen or prelogin screen

## Screenshots/Video
N/A

## Testing
Yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
New event name: vama_find_location
When to trigger: when a user clicks on the "Find a VA location" button

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx [hunk 1]
```diff
@@ -45,11 +49,18 @@ const LoginScreen: FC = () => {
...
 
-  const onFacilityLocator = navigateTo('Webview', {
-    url: WEBVIEW_URL_FACILITY_LOCATOR,
-    displayTitle: t('common:webview.vagov'),
-    loadingMessage: t('common:webview.valocation.loading'),
-  })
+  const onFacilityLocator = () => {
+    logAnalyticsEvent(Events.vama_find_location())
+    navigation.navigate(
+      'Webview' as never,
```
Comment (narin) [inline VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx:55]: Haven't seen this `as never` before.  How come it isn't needed in the simiular Webview launch in `HomeScreen.tsx` above?

Comment (dumathane) [reply VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx:55]: This has to have been an autocomplete unintended line. That's my guess

**Review by narin — APPROVED** [review]

File: VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx (hunk 1 above)
Comment (theodur) [reply VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx:43]: To fix this without using `never`, you can update the assignment of `navigation` to use the `StackNavigationProp`, like:

```typescript
 const navigation = useNavigation<StackNavigationProp<RootNavStackParamList, keyof RootNavStackParamList>>()
```

This should also type check the name of the screen that's entered in the `navigate` function

**Review by theodur — CHANGES_REQUESTED** [review]
The use of `never` can be removed by using the `StackNavigationProp` type

File: VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx (hunk 1 above)
Comment (theodur) [reply VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx:43]: To fix this without using `never`, you can update the assignment of `navigation` to use the `StackNavigationProp` type, like:

```typescript
 const navigation = useNavigation<StackNavigationProp<RootNavStackParamList, keyof RootNavStackParamList>>()
```

This should also allow for type check the name of the screen that's entered in the `navigate` function

Comment (Sparowhawk) [reply VAMobile/src/screens/auth/LoginScreen/LoginScreen.tsx]: updated

**Review by alexandec — APPROVED** [review]

**Review by theodur — DISMISSED** [review]

**Review by theodur — COMMENTED** [review]
@Sparowhawk Looks like a merge conflict just came up

**Review by alexandec — APPROVED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**